.github/workflows/release.yml

```diff
@@ -21,42 +23,106 @@ jobs:
       BUILDROOT_VERSION: '2020.08.2'
       BUILDROOT_DIR: 'buildroot'
       # Stupid hack to invalidate cache from UI
+      # https://github.com/actions/cache/issues/2#issuecomment-673493515
       CACHE_VERSION: ${{ secrets.CACHE_VERSION }}
 
     strategy:
       matrix:
         raspberry: ['raspberrypi0', 'raspberrypi0w']
 
     steps:
-      - name: Checkout code
+      - name: Checkout Code
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
 
-      # Get short SHA, as github only gives us the full SHA
-      - uses: benjlevesque/short-sha@v1.2
-        id: short-sha
+      # Retrieve annotated tags as the checkout above doesn't do this
+      # https://github.com/actions/checkout/issues/217#issuecomment-636481619
+      - name: Fetch Git Tag History
+        run: |
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          echo "SHA=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "GIT_TAG_NAME=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          echo "Current version: $(git describe)"
 
       - name: Download Buildroot
         run: |
           wget -qc https://buildroot.org/downloads/buildroot-${{ env.BUILDROOT_VERSION }}.tar.gz
           tar -zxf buildroot-${{ env.BUILDROOT_VERSION }}.tar.gz
           mv buildroot-${{ env.BUILDROOT_VERSION }} ${{ env.BUILDROOT_DIR }}
 
-      - name: Cache build
+      - name: Cache Build
         uses: actions/cache@v2
         with:
-          path: |
-            output
-            # Don't cache piwebcam as we want to force a rebuild every time
-            !output/*/build/piwebcam*
+          path: output
           key: ${{ matrix.raspberry }}-${{ env.BUILDROOT_DIR }}-${{ env.BUILDROOT_VERSION }}-${{ env.CACHE_VERSION }}
 
-      - name: Build image
+      - name: Build Image
         run: |
+          # Force a piwebcam rebuild every time
+          rm -fr output/*/build/piwebcam*
           sh build-showmewebcam.sh ${{ matrix.raspberry }}
-          mv output/${{ matrix.raspberry }}/images/sdcard.img sdcard-${{ matrix.raspberry }}-${{ env.SHA }}.img
+          cp output/${{ matrix.raspberry }}/images/sdcard.img sdcard-${{ matrix.raspberry }}-${{ env.SHA }}.img
```

Comment:
are we going back to `cp` now, why?

Reply:
yeah, because we need that file for the release. i thought it's easier to always use the base image for the right naming.